### PR TITLE
Publish OAI Pipeline component to azureml-staging registry

### PR DESCRIPTION
**Pull request by vikasagrawal-ms**
Publish OAI Pipeline component to azureml-staging registry

### CODE REVIEW

File: openai/components/oai_finetune_pipeline_component.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    type: string
+    optional: False
+    default: Ada
+    description: GPT-3 model engine
+    enum:
+      - Ada
+      - Babbage
+      - Curie
+      - DaVinci
+      - Text-Ada-001
```
Comment (nagau) [inline openai/components/oai_finetune_pipeline_component.yml:19]: Can you please sync with Bhawana for model names, we don't need inference only models in this list.

File: openai/components/oai_finetune_pipeline_component.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      - Code-Ada-002-test-stage
+  train_dataset:
+    type: uri_folder
+    optional: False
+    description: Input dataset (file or folder). If a folder dataset is passed, includes all nested files.
+  registered_model_name:
+    type: string
+    optional: False
+    description: User-defined registered model name
+  singularity_cpu_instance_type:
```
Comment (nagau) [inline openai/components/oai_finetune_pipeline_component.yml:61]: For 3P components we need serverless only.

**Review by nagau — CHANGES_REQUESTED** [review]

File: openai/components/oai_finetune_pipeline_component.yml (hunk 1 above)
Comment (vikasagrawal-ms) [reply openai/components/oai_finetune_pipeline_component.yml:19]: Done

File: openai/components/oai_finetune_pipeline_component.yml (hunk 2 above)
Comment (vikasagrawal-ms) [reply openai/components/oai_finetune_pipeline_component.yml:61]: Updating the component for 3P

File: openai/components/oai_finetune_pipeline_component.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    type: uri_folder
+    description: Dataset for the output model
+
+jobs:
+  model_selector:
+    type: command
+    component: azureml://registries/openai-dev-euap/components/microsoft_com_azureml_oai_modelselector/versions/0.1.7
+    inputs:
+      model: ${{parent.inputs.model}}
+    compute: ${{parent.inputs.singularity_subscription}}
```
[the remark this answers is not in the document]
Comment (vikasagrawal-ms) [reply openai/components/oai_finetune_pipeline_component.yml:88]: Updating the component for 3P

**Review by nagau — APPROVED** [review]

File: openai/components/oai_finetune_pipeline_component.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  model_selector:
+    type: command
+    component: azureml://registries/openai-dev-euap/components/microsoft_com_azureml_oai_modelselector/versions/0.1.7
+    inputs:
+      model: ${{parent.inputs.model}}
+
+  
+  fine_tune:
+    type: command
+    component: azureml://registries/openai-dev-euap/components/microsoft_com_azureml_oai_finetune/versions/0.1.8
```
Comment (novaturient95) [inline openai/components/oai_finetune_pipeline_component.yml:50]: is this a publicly accessible registry?

Comment (vikasagrawal-ms) [reply openai/components/oai_finetune_pipeline_component.yml:50]: No.

File: openai/components/oai_finetune_pipeline_component.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+version: 0.0.1
+type: pipeline
+display_name: PipelineComponent for OpenAI Finetuning workflow
+description: Pipeline component for OpenAI finetuning workflow
+
+settings:
+  default_compute: azureml:serverless
+
+inputs:
+  model:
```
Comment (novaturient95) [inline openai/components/oai_finetune_pipeline_component.yml:12]: nit: mode_id?

Comment (vikasagrawal-ms) [reply openai/components/oai_finetune_pipeline_component.yml:12]: no it takes model name as string input and internally resolves to model asset id

Comment (vikasagrawal-ms) [reply openai/components/oai_finetune_pipeline_component.yml:12]: no it takes model name as string input and internally resolves to model asset id

**Review by novaturient95 — APPROVED** [review]

File: openai/components/oai_finetune_pipeline_component.yml (hunk 4 above)
Comment (novaturient95) [reply openai/components/oai_finetune_pipeline_component.yml:50]: It would fail then, right?

Comment (novaturient95) [reply openai/components/oai_finetune_pipeline_component.yml:50]: It would fail then, right?

Comment (vikasagrawal-ms) [reply openai/components/oai_finetune_pipeline_component.yml:50]: It will not, as it will be running on IPP infra.

**Review by vizhur — APPROVED** [review]

**Pull request merged**